modules/integration/tests-ui-integration/tests-ui/src/test/java/org/wso2/ds/ui/integration/test/dashboard/DashboardAnonViewTest.java

```diff
@@ -268,17 +165,18 @@ public void testAnonDashboardPages() throws Exception {
     @Test(groups = "wso2.ds.dashboard", description = "Remove anonymous view mode from added dashboard page in dashboard",
             dependsOnMethods = "testAnonDashboardPages")
     public void testAnonDashboardPageRemove() throws Exception {
-        String defaultGadgetExpected = "G2";
-
+        switchPage("page0");
         getDriver().findElement(By.cssSelector("input[name='anon']")).click();
         boolean isToggleButtonHidden = getDriver().findElements(By.cssSelector(".toggle-design-view.hide")).size() > 0;
-        String defaultGadgetActual = getDriver().findElement(By.cssSelector("iframe")).getAttribute("title");
-        getDriver().get(getBaseUrl() + "/portal/dashboards/" + dashboardTitle + "/landing?isAnonView=true");
-        boolean isLinkForNonAnonPageNotExist = getDriver().findElements(By.cssSelector("a[href='" + getBaseUrl() +
-                "/portal/dashboards/" + dashboardTitle + "/page0?isAnonView=true']")).size() <= 0;
 
-        assertEquals(defaultGadgetExpected, defaultGadgetActual);
         assertTrue(isToggleButtonHidden, "Anonymous toggle button is not hidden");
+        assertEquals(GADGET_3, getAttributeValue("iframe", "title"));
+
+        redirectToLocation("portal", "dashboards/" + dashboardTitle + "/landing?isAnonView=true");
```

Comment:
Export the commonly used URIs to a constant class, and reuse in all the locations.